### PR TITLE
HPCC-24080 add zip package to Docker base image

### PR DESCRIPTION
**Pull request by xwang2713**
Install zip package for zip executable.

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

@dcamper please review

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by dcamper — APPROVED** [review]
@xwang2713 Looks good.

File: dockerfiles/platform-build-base/Dockerfile [hunk 1]
```diff
@@ -67,6 +67,7 @@ RUN apt-get install -y \
   rsync \
   psmisc \
   libpopt0 \
+  zip \
```
Comment (richardkchapman) [inline dockerfiles/platform-build-base/Dockerfile:70]: I think this is the wrong Dockerfile - you want this at runtime, not a build time.

Comment (xwang2713) [reply dockerfiles/platform-build-base/Dockerfile:70]: OK

File: dockerfiles/platform-core/Dockerfile [hunk 2]
```diff
@@ -39,6 +39,7 @@ RUN apt-get install -y \
   libapr1 \
   python \ 
   psmisc \
+  zip \
```
Comment (richardkchapman) [inline dockerfiles/platform-core/Dockerfile:42]: I may be wrong - the structure of the debug-related Dockerfiles is a bit confusing - but I think you also need it in platform-core-debug

Comment (xwang2713) [reply dockerfiles/platform-core/Dockerfile:42]: Right. Add zip to platform-core-debug/Dockerfile

**Comment by richardkchapman** [comment]
@xwang2713 Please squash

**Comment by xwang2713** [comment]
Rebased

**Pull request merged**